http3/transport.go

```diff
@@ -258,7 +263,10 @@ func (t *Transport) getClient(ctx context.Context, hostname string, onlyCached b
 		go func() {
 			defer close(cl.dialing)
 			defer cancel()
+			traceConnectStart(trace, "udp", hostname)
 			conn, rt, err := t.dial(ctx, hostname)
+			traceConnectDone(trace, "udp", hostname, err)
+			traceTLSHandshakeStart(trace)
```

Comment:
This is not correct. The TLS handshake is part of the QUIC handshake. You probably want to exclude address resolution though.

Reply:
I also have some doubts about this part, because the QUIC handshake includes the TLS handshake, which is completely different from TCP + TLS. I referred to the implementation of curl, and the TLS handshake time is generally around 1-2 ms. I understand that this should not include network overhead; it is just a local processing delay. 

Here, I would also like to hear your suggestions.

Reply:
I looked at the implementation of the standard library, and I understand that the time taken by the `sysDialer.dialSingle` function corresponding to `ConnectStart` and `ConnectDone` can be related to the time taken by the `quic.Transport.dial` function.

Reply:
The TLS handshake should only be traced when we're actually establishing a new QUIC connection. That means that in the case of a reused connection, we don't call it at all.

Reply:
I gave up. In QUIC, it seems that the connection establishment and TLS handshake cannot be accurately separated; they are both included in the QUIC handshake process.

Reply:
You shouldn't separate it. Just emit the event at the same time you emit the `ConnectStart` event. That actually is **exactly** the right thing to do.

Same goes for `ConnectDone` and `TLSHandshakeDone` (be careful of 0-RTT though).